js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/QueryOptionsTab.tsx

```diff
@@ -115,7 +115,7 @@ export const QueryOptionsTab = ({ mode }: { mode: "json" | "table" }) => {
           };
           if (!errors.length) {
             setQueryOptions(updatedOptions || {});
-            setDisabledRules(updatedValue.optimizer?.rules);
+            setDisabledRules(updatedValue.optimizer?.rules || []);
```

Comment:
do you know why typescript didn't show this issue? probably because `any` was used there instead of `unknown`; just explore if we can fix the type as well

Reply:
let's see:
`setDisabledRules` is already typed with `React.Dispatch<React.SetStateAction<string[]>>`
The `any` we are getting here on the onChange is defined at [jsoneditor-react.d.ts](https://github.com/arangodb/arangodb/blob/4756b943e3998b565bd2f2fc2ddd5809714f7a55/js/apps/system/_admin/aardvark/APP/react/jsoneditor-react.d.ts)

We could use generics, but `ControlledJSONEditor` internally uses `forwardRef`, and `forwardRef` is not "generics friendly" so I am forced to redefine the forwardRef typing (it is part of my proposed solution).

So to contain the change, since it is a bug fix I've tried to improve type safety without going too far outside the scope of this PR / without refactoring too many things.

please review.

Reply:
@palashkaria

Reply:
Looks good!


A note for later:

We do have `forwardRef` with generics being used in the UI lib here: https://github.com/arangodb/ui/blob/810c2bf11acac7a485adf550fa1186e195cb58c0/packages/ui/src/Select/SelectBase.tsx#L24

Can you take a look at that? We can integrate your approach there maybe, & another suggestion is to move the `ControlledJSONEditor` component into the UI lib later & keep these module decleration in one place @cmyk47

Reply:
Yes, I think we can simplify it, let's keep that in mind for the future.